### ISSUE TITLE
netlify-lambda serve creates additional files in functions folder

### ISSUE THREAD

**Issue by gtalarico** [netlify/netlify-lambda#162]
*TLDR*
`netlify-lambda serve` copies project files into target `functions` folder

I have a single function called `airtable.js` in a folder called `functions`.

The function is runs fine locally, but calling `netlify-lambda serve .` also creates additional files inside the `functions` folder. If I leave those files there the build fails.

As workaround, I can gitignore those files, but seems strange to me they are created 
every time I run the cmd.

```
#netlify.toml
[build]
  functions = "functions"
```
### Project
```

├── README.md
├── components/
├── dataService.js
├── functions
│   └── airtable.js
│
├── netlify.toml
├── nuxt.config.js
├── package.json
├── pages/
├── utils.js
└── yarn.lock
```

I then run the cmd below to serve the functions
```
yarn netlify-lambda serve .
```

After running, the directory now looks like this. Note the extra project files in the `functions/` folder:
```
├── README.md
├── components/
├── dataService.js
├── functions
│   ├── airtable.js
│   ├── dataService.js
│   ├── nuxt.config.js
│   └── utils.js
│
├── netlify.toml
├── nuxt.config.js
├── package.json
├── pages/
├── utils.js
└── yarn.lock
```

I am not sure why `netlify-lambda` is pulling those JS files and copying them into the functions folder...

![image](https://user-images.githubusercontent.com/9513968/59005229-90470980-87d1-11e9-97e2-bcb959f1f865.png)

**Comment by sw-yx** [comment]
this is happening because you have the wrong mental model of how `netlify-lambda` works. it takes a source folder, and builds it to a destination folder. the source folder is specified in your command, in your case `netlify-lambda serve .`, which means your source folder is the root dir of your project. the destination folder is the one you specified in your destination folder, i.e. `functions`.

the correct set up you probably want requires two changes to what you're doing.

1. modify the destination folder to something else, e.g. `functions = "functions-built"` inside your `netlify.toml`.
2. modify your serve command to point to the right source folder: `netlify-lambda serve functions`.

have a look at our reference https://github.com/netlify/create-react-app-lambda demo to see this at work

**Comment by gtalarico** [comment]
It was working local and deploy, so I thought I had it right - I obviously didn't :)
Going back to the docs and re-read them now.
Thank you for your help
